packages/components/src/components/progress-indicator/_progress-indicator.scss

```diff
@@ -70,6 +80,11 @@
     }
   }
 
+  .#{$prefix}--progress--space-equal .#{$prefix}--progress-label {
+    max-width: 100%;
```

Comment:
Because I was looking into this feature as well; be aware that the hover state for the steps are a little bit off:
![image](https://user-images.githubusercontent.com/3808948/76003544-7f662880-5f08-11ea-9608-e94de36267c8.png)